### PR TITLE
Make log stream configurable

### PR DESCRIPTION
**Pull request by hoegaarden**
Keeps the default behaviour to log to stdout.

Introduces a new config option `LOG_STREAM`, which can be set to either `STDOUT` or `STDERR` (where the default is the former). If it's set to something different, it will fallback to the logging library's default.

### CODE REVIEW

**Comment by hoegaarden** [comment]
Related to #89 & #90

File: deye_docker_entrypoint.py [hunk 1]
```diff
@@ -20,18 +20,20 @@
...
-
-logging.basicConfig(stream=sys.stdout, format=log_format, level=logging.getLevelName(config.log_level))
-
-log = logging.getLogger('main')
+def setupLogging(config):
+    log_format = "%(asctime)s - %(name)s - %(levelname)s - %(message)s"
 
+    log_stream = None
+    if config.log_stream == LOG_DEST_STDOUT : log_stream = sys.stdout
+    if config.log_stream == LOG_DEST_STDERR : log_stream = sys.stderr
```
Comment (kbialek) [inline deye_docker_entrypoint.py:30]: Please add basic validation here. My expectation is that the process will exit with status 1 if `config.log_stream` is set to unexpected value.

**Review by kbialek — CHANGES_REQUESTED** [review]

File: deye_docker_entrypoint.py (hunk 1 above)
Comment (hoegaarden) [reply deye_docker_entrypoint.py]: Done.
PTAL

File: deye_docker_entrypoint.py [hunk 2]
```diff
@@ -20,18 +20,28 @@
...
+def setupLogging(config):
+    log_format = "%(asctime)s - %(name)s - %(levelname)s - %(message)s"
+
+    supported_log_streams = {
+            LOG_DEST_STDOUT: sys.stdout,
+            LOG_DEST_STDERR: sys.stderr,
+            }
+    if config.log_stream not in supported_log_streams.keys():
+        err = f"Unsupported log stream {config.log_stream}, supported: {','.join(supported_log_streams.keys())}"
+        raise ValueError(err)
```
Comment (kbialek) [inline deye_docker_entrypoint.py:34]: Where is this exception handled? IMO, the process should print an error and exit. No exception traceback is needed.

**Review by kbialek — CHANGES_REQUESTED** [review]

File: deye_docker_entrypoint.py (hunk 2 above)
Comment (hoegaarden) [reply deye_docker_entrypoint.py:34]: The exception is not handled, but _could_ be handled if that thing was consumed by something different.
This behaves similar as if you would set `LOG_LEVEL` to something unknown/unsupported.

I am however happy to change that, do a print & sys.exit or something.

Comment (kbialek) [reply deye_docker_entrypoint.py:34]: > This behaves similar as if you would set `LOG_LEVEL` to something unknown/unsupported.

Valid point. Then we should wrap call to setupLogging with try...except at line 43 and print a meaningful error message when logger config is incorrect. WDYT?

Comment (hoegaarden) [reply deye_docker_entrypoint.py]: Sounds reasonable to me.
I rebased on the new version of `main` and (I think) implemented what you suggested.
PTAL.

**Comment by kbialek** [comment]
Please add the new env variable in README.md and also in `config.env.example`

File: src/deye_config.py [hunk 3]
```diff
@@ -127,6 +131,7 @@ def from_env():
             DeyeLoggerConfig.from_env(),
             DeyeMqttConfig.from_env(),
             log_level=os.getenv("LOG_LEVEL", "INFO"),
+            log_stream=os.getenv("LOG_STREAM", "LOG_DEST_STDOUT"),
```
[the remark this answers is not in the document]
Comment (hoegaarden) [reply src/deye_config.py]: Sorry ... pushed to fast.

**Review by kbialek — APPROVED** [review]
:+1:

**Comment by kbialek** [comment]
@hoegaarden Your code violates linting rules. Reformat it with black formatter. See CONTRIBUTING.md https://github.com/hoegaarden/deye-inverter-mqtt/actions/runs/5181981700/jobs/9338212828

**Comment by hoegaarden** [comment]
> @hoegaarden Your code violates linting rules. Reformat it with black formatter. See CONTRIBUTING.md https://github.com/hoegaarden/deye-inverter-mqtt/actions/runs/5181981700/jobs/9338212828

Sorry, not too familiar with the whole python ecosystem. Relied on the default config of my editor, which probably was not the best choice.
However, should be fixed now.

**Comment by kbialek** [comment]
> Sorry, not too familiar with the whole python ecosystem and its conventions and tooling. Relied on the default config of my editor, which probably was not the best choice.
> However, should be fixed now.

No worries. This is a new requirement. It's turned out that more people wants to contribute to this project, so I had to put this kind of restrictions to keep the code in shape.

**Pull request merged**